packages/wrangler/src/api/pages/deploy.tsx

```diff
@@ -139,15 +147,26 @@ export async function deploy({
 		isProduction = project.production_branch === branch;
 	}
 
-	const deploymentConfig =
-		project.deployment_configs[isProduction ? "production" : "preview"];
+	const env = isProduction ? "production" : "preview";
+	const deploymentConfig = project.deployment_configs[env];
 	const nodejsCompat =
 		deploymentConfig.compatibility_flags?.includes("nodejs_compat") ?? false;
 
 	const defineNavigatorUserAgent = isNavigatorDefined(
 		deploymentConfig.compatibility_date,
 		deploymentConfig.compatibility_flags
 	);
```

Comment:
These values should come from the config file, if possible

Reply:
you mean 👇 ?

```
const nodejsCompat =config?.compatibility_flags?.includes("nodejs_compat") ??
  ?? deploymentConfig.compatibility_flags?.includes("nodejs_compat")
  ?? false;

const defineNavigatorUserAgent = isNavigatorDefined(
		config.compatibility_date ?? deploymentConfig.compatibility_date,
		config?.compatibility_flags ?? deploymentConfig.compatibility_flags
	);
```

Reply:
fixed